src/utils.rs

```diff
@@ -901,14 +901,30 @@ pub fn rewrite_urls(input_text: &str) -> String {
 			let formatted_url = format_url(REDDIT_PREVIEW_REGEX.find(&text1).map(|x| x.as_str()).unwrap_or_default());
 
 			let image_url = REDLIB_PREVIEW_LINK_REGEX.find(&formatted_url).map_or("", |m| m.as_str()).to_string();
-			let image_text = REDLIB_PREVIEW_TEXT_REGEX.find(&formatted_url).map_or("", |m| m.as_str()).to_string();
+			let mut image_text = REDLIB_PREVIEW_TEXT_REGEX.find(&formatted_url).map_or("", |m| m.as_str()).to_string();
+			if !image_text.is_empty() {
+				image_text.remove(0);
+				image_text.pop();
+				image_text.pop();
+				image_text.pop();
+				image_text.pop();
+			}
```

Comment:
Not sure exactly what this does but it should probably operate on a substring like `image_text[1..image_text.len()-4]` - if what you intend to do is remove the last 4 and first. Curious why this is done like this - can you think of a better way to do this?

Reply:
The first character of `image_text` is a `>`, and could stay, but it would require readding the `.replace(">>", ">")` to `image_to_replace`'s assignment since `image_url` also contains a `>` at the end of it meaning we end up with `>>` in between `{image_url}` and `{image_text}` instead of just a single `>`. I figured just removing the first character is better than theoretically having the potential to replace an intentional >> somewhere else inside of `image_text` and having the replacement fail.

The last four need to be removed since they'll be an `</a>`, which we don't want appearing inside of the `<figcaption>` block. Although now that I think about it that can just be done within the if block since it isn't needed otherwise.

I'll modify it to use the substring instead and only drop the `</a>` when needed.

Reply:
Just made it use the substring as well as renaming `image_text` to `image_caption` and adding comments to make thing's functions and usages clearer.

I didn't make it so that the `</a>` is only dropped when `image_caption` is actually used to add a caption to the image as it seemed like an unnecessary extra step when it can be done at the same time as removing the first character.
However I did make it so that the quotes inside `image_caption` are only replaced when it's actually used as a caption since it wasn't needed otherwise.